src/transport/http/server/background.rs

```diff
@@ -166,8 +166,14 @@ async fn process_request(
 
             let json_body = match body_to_request(request.into_body()).await {
                 Ok(b) => b,
-                Err(_) => {
-                    unimplemented!() // TODO:
+                Err(e) => {
+                    use std::io::ErrorKind::*;
+                    match (e.kind(), e.into_inner()) {
+                        (InvalidData, _) => return response::parse_error(),
+                        (UnexpectedEof, _) => return response::parse_error(),
```

Comment:
This is very subjective, but could you remove importing `*`, and imports within the body of functions?

```suggestion
                    match (e.kind(), e.into_inner()) {
                        (io::ErrorKind::InvalidData, _) => return response::parse_error(),
                        (io::ErrorKind::UnexpectedEof, _) => return response::parse_error(),
```